executor/linux_fork_server_executor.cpp

```diff
@@ -0,0 +1,585 @@
+    // Although, PUT process should be handled by fork server, kill it just in case. (Since fork server is expected to be killed using kill, therefore it will never becoome a zombie. So never wait.)
+    KillChildWithoutWait();
+}
+
+void LinuxForkServerExecutor::SetCArgvAndDecideInputMode() {
+    assert(!argv.empty()); // It's incorrect too far
+
+    stdin_mode = true; // if we find @@, then assign false to stdin_mode
+
+    for (const auto& v : argv ) {
+        if ( v == "@@" ) {
+            stdin_mode = false;
+            cargv.emplace_back(path_str_to_write_input.c_str());
+        } else {
+            cargv.emplace_back(v.c_str());
+        }
+    }
+    cargv.emplace_back(nullptr);
+}
+
+void LinuxForkServerExecutor::TerminateForkServer() {
+    put_channel.TerminateForkServer();
+}
+
+/**
+ * An static method
+ * Precondition:
+ *  - active_instance has a non-nullptr value definitely when this function is called.
+ *  - active_instance has an address of valid Executor instance.
+ *  - The global timer exists in the self process, and the function is called for SIGALRM signal.
+ * Postcondition:
+ *  - The function works as signal handler for SIGALRM
+ *      - If called, the PUT is considered as expiring the execution time, therefore kill active_instance->child_pid.
+ *      - Set a flag active_instance->child_timed_out
+ */
+void LinuxForkServerExecutor::AlarmHandler(int signum) {
+    assert (signum == SIGALRM);
+    assert (active_instance != nullptr);
+    active_instance->KillChildWithoutWait();
+    active_instance->child_timed_out = true;
+}
+
+/*
+ * An static method
+ * Postcondition:
+ *  - It defines how the fuzzuf process respond to signals.
+ *  - If signal handlers are needed, signal handlers are set.
+ * FIXME: Since the configuration on signal handlers is shared in whole process,
+ * it affects all fuzzuf instances running on the process.
+ * As the result, it requires change of process design in some cases,
+ * Therefore, the process design may be changed, 
+ * yet currently the static function is implemented under the expection that only one fuzzuf instance is used simultaneously.
+ */
+void LinuxForkServerExecutor::SetupSignalHandlers() {
+    struct sigaction sa;
+
+    sa.sa_handler = NULL;
+    sa.sa_flags = SA_RESTART;
+    sa.sa_sigaction = NULL;
+
+    sigemptyset(&sa.sa_mask);
+// Since there is no agreement on handling following signals, ignoring for now.
+#if 0
+    /* Various ways of saying "stop". */
+
+    sa.sa_handler = handle_stop_sig;
+    sigaction(SIGHUP, &sa, NULL);
+    sigaction(SIGINT, &sa, NULL);
+    sigaction(SIGTERM, &sa, NULL);
+
+    /* Window resize */
+
+    sa.sa_handler = handle_resize;
+    sigaction(SIGWINCH, &sa, NULL);
+
+    /* SIGUSR1: skip entry */
+
+    sa.sa_handler = handle_skipreq;
+    sigaction(SIGUSR1, &sa, NULL);
+#endif
+
+    /* Things we don't care about. */
+
+    sa.sa_handler = SIG_IGN;
+    sigaction(SIGTSTP, &sa, NULL);
+    sigaction(SIGPIPE, &sa, NULL);
```

Comment:
I think this function is also no longer needed, but maybe these signals should be handled in somewhere else like CLIEntryPoint?

Reply:
Thank you for suggestion. I removed these code.